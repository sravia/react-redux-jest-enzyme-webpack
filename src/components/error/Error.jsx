import React from 'react';
import PropTypes from 'prop-types';

const Error = ({ error }) => (
  <div>
    <span>{error}</span>
  </div>
);

Error.propTypes = {
  error: PropTypes.string.isRequired
};

export default Error;
