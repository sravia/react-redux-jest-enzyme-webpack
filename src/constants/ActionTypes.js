export const GET_RATES_REQUEST = 'GET_RATES_REQUEST';
export const GET_RATES_SUCCESS = 'GET_RATES_SUCCESS';
export const GET_RATES_FAILURE = 'GET_RATES_FAILURE';
