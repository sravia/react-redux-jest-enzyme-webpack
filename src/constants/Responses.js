export const API_ERROR = 'Could not fetch latest rates!';
export const TEST_API_RESPONSE = {
  date: '2018-10-23',
  rates: {
    BGN: 1.9558,
    CAD: 1.5047,
    BRL: 4.2412,
    HUF: 322.81,
    DKK: 7.4603,
    JPY: 128.76,
    ILS: 4.2086,
    TRY: 6.6227,
    RON: 4.667,
    GBP: 0.88103,
    PHP: 61.716,
    HRK: 7.4275,
    NOK: 9.5053,
    USD: 1.1478,
    MXN: 22.2469,
    AUD: 1.6243,
    IDR: 17435.02,
    KRW: 1305.81,
    HKD: 8.999,
    ZAR: 16.4759,
    ISK: 136.3,
    CZK: 25.814,
    THB: 37.705,
    MYR: 4.7778,
    NZD: 1.7514,
    PLN: 4.296,
    SEK: 10.3606,
    RUB: 75.0301,
    CNY: 7.9613,
    SGD: 1.5822,
    CHF: 1.1424,
    INR: 84.4435
  },
  base: 'EUR'
};
