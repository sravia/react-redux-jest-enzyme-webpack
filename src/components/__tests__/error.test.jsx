import React from 'react';
import renderer from 'react-test-renderer';
import { API_ERROR } from '../../constants/responses';
import Error from '../error/Error';

describe('<Error />', () => {
  it('renders <Error /> component', () => {
    const tree = renderer.create(<Error error={API_ERROR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
