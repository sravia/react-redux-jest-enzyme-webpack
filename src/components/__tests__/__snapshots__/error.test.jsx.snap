// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> renders <Error /> component 1`] = `
<div>
  <span>
    Could not fetch latest rates!
  </span>
</div>
`;
